### ISSUE TITLE
OpenToonz Places Full Screen Mode Drawing Area on Wrong Monitor

### ISSUE THREAD

**Issue by BloodyMess** [opentoonz/opentoonz#3700]
**Describe the bug**
I have OpenToonz maximized on my preferred display (Gaomon PD2200 Pen Display). When I choose "Full Screen Mode" from the contextual menu, it places the the drawing content area maximized on my other monitor, which is in portrait orientation. This is useless because then I can't use the pen display to draw on it.

**To Reproduce**
List steps to reproduce the behavior.
1. Launch OpenToonz
2. Maximize window on main display.
3. Choose "Full Screen Mode" from the contextual menu.

**Expected behavior**
I expect the full screen drawing area to simply stay on the same monitor that the window is already on.

**Screenshots**
Here is how my monitors are laid out:
![Screenshot from 2021-01-29 07-08-07](https://user-images.githubusercontent.com/12903370/106291897-0d25c300-6201-11eb-8ae0-d19a9bbf9fc5.png)

Here is how OpenToonz is putting the full screen drawing area on the wrong monitor:
![Screenshot from 2021-01-29 07-12-37](https://user-images.githubusercontent.com/12903370/106292306-858c8400-6201-11eb-95ba-bc5be2dfa450.png)

**Device Information:**
 - OS: elementary OS 5.1.7 Hera (based on Ubuntu 18.04 LTS)
 - Kernel: Linux 5.4.0-65-generic
 - GTK Version: GTK 3.22.30
 - OpenToonz Version: Built from current source checked out from master branch.
 - GPU: Sapphire Pulse Radeon RX590 8GB
 - GPU Driver Version: "xserver-xorg-video-amdgpu-hwe-18.04/bionic-updates,now 19.1.0-1~18.04.1 amd64 [installed,automatic]"

**Comment by RodneyBaker** [comment]
This seems likely related to an issue where multiple monitors must be of the same resolution.

You might test that theory by making your screens mirror each other and see if that resolves the issue.

Some users have reported that tweaking their graphics settings and drivers has resolved this problem but in general the issue seems to persist with likely issues upstream related to the use of Qt and graphics cards.

**Comment by BloodyMess** [comment]
Nope. No luck for me with mirroring or with just changing orientation.

I did notice something while testing your suggestions. And that is that OpenToonz is always using the leftmost monitor for its "full screen" mode. On a hunch, I configured the portrait monitor to be on the right-hand side of the main (pen display) monitor. That did the trick! Unfortunately, actually moving the monitor to match that layout isn't an option for me. So that work-around isn't viable for me.

My guess is that OpenToonz is assuming that "full screen" mode means that the full screen window is supposed to be located at coordinates (0,0), or at least that the x coordinate is supposed to be 0.

One other bug I noticed while doing this testing: When the displays are arranged so the main display is below the secondary display, I can't bring up any menu. Both the menubar menus and the contextual menu won't display. I get mouse-over feedback in the menubar, but clicking to display a menu doesn't work.

**Comment by BloodyMess** [comment]
> but in general the issue seems to persist with likely issues upstream related to the use of Qt and graphics cards.

I looked at the OpenToonz source, and in `MainWindow::fullScreenWindow()`, it's just calling the Qt method `QWindow::setWindowState()` with `Qt::WindowFullScreen` as the parameter.

Seeing that, I searched for anyone else having problems getting Qt to give them the full-screen behavior they want. [This discussion on the Qt forums](https://forum.qt.io/topic/66935/fullscreen-over-multiple-screens-qt-5-6) seems to be relevant, especially the last four posts. It sounds like it might be possible to borrow the approach described there to manually implement a better full-screen mode that works with multi-monitor systems.

**Comment by BloodyMess** [comment]
@RodneyBaker I took my own suggestion and implemented it.

I was right about Qt always putting the window at (0,0), but I was initially looking at the wrong class and method. I implemented the fix in FullScreenWidget::toggleFullScreen() in [imageutils.cpp in my fork](https://github.com/BloodyMess/opentoonz/blob/master/toonz/sources/toonzqt/imageutils.cpp).

It seems to be working fine, but I only have a Linux system to test it on. So I'm hesitant to create a pull request. Also, my coding style is a bit different from what's in OpenToonz, you might prefer to just look at my fix to see what I did (don't worry, it's super easy to read and very well commented.).
